exercises/difference-of-squares/src/example.h

```diff
@@ -0,0 +1,8 @@
+#ifndef _DIFFERENCE_OF_SQUARES_H
+#define _DIFFERENCE_OF_SQUARES_H
+
+unsigned int sum_of_squares(unsigned int number);
+unsigned int square_of_sum(unsigned int number);
+unsigned int difference(unsigned int number);
```

Comment:
I was trying to follow other tracks, most of which use `difference`.  But in our case, since we don't have a `Squares` class, I agree that `difference_of_squares` is better.